### PR TITLE
Extract LdapOrgReaderProcessor into a separate package

### PR DESCRIPTION
**Pull request by goober**
Signed-off-by: Mathias Åhsberg <mathias.ahsberg@resurs.se>

## Hey, I just made a Pull Request!

This pull request is heavily inspired by the extraction of the MicrosoftGraph processor in #5854 and follows the same pattern. 

The final goal is to add support for customized transformers just as the MicrosoftGraph processor. However, To make it easier to review I want to split it into two separate parts.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by goober** [comment]
Related to #5221

**Comment by freben** [comment]
Thanks for getting this started!

Idea for a future, separate PR:

We should totally add a `@backstage/plugin-catalog-backend-node` package as well as per [the ADR](https://backstage.io/docs/architecture-decisions/adrs-adr011), where common types and shared needs can live. So we can get rid of the copied code that you mention yourself, and also so that modules can depend on that package instead of the plugin package itself which kinda feels like a backwards dep at this point.

**Comment by benjdlambert** [comment]
@goober apologies looks like theres a few files here that have conflicts. Could be to do with the copyright headers which have now changed to be `Backstage Authors` instead of `Spotify`

**Comment by goober** [comment]
> @goober apologies looks like theres a few files here that have conflicts. Could be to do with the copyright headers which have now changed to be `Backstage Authors` instead of `Spotify`

Thank you @benjdlambert I have updated the licence headers.

File: plugins/catalog-backend-module-ldap/src/ldap/org.ts [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GroupEntity, UserEntity } from '@backstage/catalog-model';
+
+// TODO: Copied from plugin-catalog-backend, but we could also export them from
+// there. Or move them to catalog-model.
```
Comment (freben) [inline plugins/catalog-backend-module-ldap/src/ldap/org.ts:20]: yeah let's keep it here for now 👍

**Review by freben — COMMENTED** [review]
Nice! Please let's just remove the extra changeset row, and then we :shipit:, good job!

File: .changeset/tricky-carrots-lick.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+---
+'@backstage/plugin-catalog-backend': minor
+'@backstage/plugin-catalog-backend-module-ldap': patch
```
[the remark this answers is not in the document]
Comment (goober) [reply .changeset/tricky-carrots-lick.md]: Fixed

**Review by freben — APPROVED** [review]

**Pull request merged**